src/inc/til/coalesce.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til
+{
+    // Method Description:
+    // - Base case provided to handle the last argument to CoalesceOptionals<T...>()
+    template<typename T>
+    T CoalesceOptionals(const T& base)

```

Comment:
The naming of these doesn't really fit the naming theme of other `til` things. Primarily the all lowercase with underscores if necessary, concise/short as possible names, and reusing terminology/monikers from STL where possible.

Maybe `til::coalesce` for `til::CoalesceOptionals` much like `std::optional<T>::value()` with the partner
`til::coalesce_or_not` or `til::coalesce_or_empty` much like `std::optional<T>::value_or(T)`

Reply:
maybe `til::coalesce(opt, opt, opt)` and `til::coalesced_value(opt, opt, opt, val)`. you can even rebuild coalesced_value on top of coalesce!